https://bugs.gentoo.org/901809
https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2935

From 28a599631de8fbbf28777e5238da1d7addcc0d56 Mon Sep 17 00:00:00 2001
From: Chris Mayo <aklhfex@gmail.com>
Date: Sat, 1 Apr 2023 18:02:56 +0100
Subject: [PATCH] backend: Fix build with -Dremote_desktop=false
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

../mutter-44.0/src/backends/meta-backend.c: In function ‘meta_backend_real_post_init’:
../mutter-44.0/src/backends/meta-backend.c:560:7: error: ‘MetaBackendPrivate’ {aka ‘struct _MetaBackendPrivate’} has no member named ‘remote_access_controller’
  560 |   priv->remote_access_controller =
      |       ^~

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/2655
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2935>
---
 src/backends/meta-backend.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/backends/meta-backend.c b/src/backends/meta-backend.c
index d278e6a85..e77b9612b 100644
--- a/src/backends/meta-backend.c
+++ b/src/backends/meta-backend.c
@@ -63,6 +63,7 @@
 #include "backends/meta-input-settings-private.h"
 #include "backends/meta-logical-monitor.h"
 #include "backends/meta-monitor-manager-dummy.h"
+#include "backends/meta-remote-access-controller-private.h"
 #include "backends/meta-settings-private.h"
 #include "backends/meta-stage-private.h"
 #include "backends/x11/meta-backend-x11.h"
@@ -77,7 +78,6 @@
 #include "meta/util.h"
 
 #ifdef HAVE_REMOTE_DESKTOP
-#include "backends/meta-remote-access-controller-private.h"
 #include "backends/meta-remote-desktop.h"
 #include "backends/meta-screen-cast.h"
 #endif
@@ -134,8 +134,8 @@ struct _MetaBackendPrivate
 #endif
   MetaSettings *settings;
   MetaDbusSessionWatcher *dbus_session_watcher;
-#ifdef HAVE_REMOTE_DESKTOP
   MetaRemoteAccessController *remote_access_controller;
+#ifdef HAVE_REMOTE_DESKTOP
   MetaScreenCast *screen_cast;
   MetaRemoteDesktop *remote_desktop;
 #endif
@@ -1410,13 +1410,9 @@ meta_backend_get_screen_cast (MetaBackend *backend)
 MetaRemoteAccessController *
 meta_backend_get_remote_access_controller (MetaBackend *backend)
 {
-#ifdef HAVE_REMOTE_DESKTOP
   MetaBackendPrivate *priv = meta_backend_get_instance_private (backend);
 
   return priv->remote_access_controller;
-#else
-  return NULL;
-#endif
 }
 
 /**
-- 
2.39.2

